Make code work with one shipment per order (even if the DB still supports Spree's multi shipments per order model)

luisramos0 opened the pull request:
#### What? Why?

This is a draft attempt at #6428 

I thought this would not be obvious for other people, this is what I think we can do.

Even better would be to change has_many :shipments in order.rb to has_one :shipment but I dont know how to do that without changing the DB? (the spree_shipments table with an order_id and no shipment_id in the spree_orders table)

Even better would be to change the DB... :-) 


#### What should we test?
<!-- List which features should be tested and how. -->



#### Release notes
<!-- Write a one liner description of the change to be included in the release notes.
Every PR is worth mentioning, because you did it for a reason. -->

<!-- Please select one for your PR and delete the other. -->
Changelog Category: Technical changes
Making OFN code simpler in the shipments domain.

luisramos0 committed:
MAke more code just use the single shipment of the order even if the DB still supports multiple shipments per order

luisramos0 committed:
Make code work with a single package per order even if the DB dtill supports multiple shipments per order in OFN, one order only has one shipment

luisramos0 commented:
ping @Matt-Yorkley I wonder if you are looking at similar things as you look at adjustments. I am not sure if I have pinged you already, you will like this draft.

Matt-Yorkley commented:
Nice :+1: I think I found an upstream commit from 2-2-stable which also would improve loading the correct selected shipping rate.

luisramos0 committed:
Remove order shipment concern

Matt-Yorkley committed:
Fix some pluralisation issues on `package` variables

Matt-Yorkley commented:
I think we should have a go at this. I need to start touching shipments elsewhere, and it'll be simpler if we do this first. I'll take a look at it, there's currently a few build errors.

Matt-Yorkley committed:
Fix #prioritized_package issue

#empty? is used on enumerators, it's not plural now.

Matt-Yorkley committed:
Fix expected boolean responses in #backordered? and #ready_to_ship?

These methods need to explicitly return false if there's no shipment (as they did before).

Matt-Yorkley committed:
Delete OrderShipment concern spec

Matt-Yorkley commented:
It feels like some of these obscure classes like `Stock::Prioritizer` are designed solely to handle things like packages being split across multiple shipments...?

luisramos0 commented:
This stock code is actually the best code in spree imo. Yes, it is mostly
useless for ofn. It would require a bit more work to remove.

On Sun, 7 Feb 2021, 12:23 Matt-Yorkley, <notifications@github.com> wrote:

> It feels like some of these obscure classes like Stock::Prioritizer are
> designed solely to handle things like packages being split across multiple
> shipments...?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openfoodfoundation/openfoodnetwork/pull/6433#issuecomment-774665833>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMQPOTMRTFTK37JQKRYWGDS52A3PANCNFSM4UICFVMA>
> .
>

andrewpbrett commented:
I'm going to move this back to the Tech Debt column unless we're still actively working on it, feel free to move it back if so.

luisramos0 commented:
ok, I am not planning to pick this up now but it's still a valid PR that makes the core order code easier to work with.
